components/brave_rewards/resources/ui/components/mobile/promoMobile/promo_mobile.tsx

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import * as style from './promo_mobile.style'
+import { CloseStrokeIcon } from 'brave-ui/components/icons'
+import { LocaleContext } from '../../../../shared/lib/locale_context'
+
+export interface Props {
+  title: string
+  imagePath: string
+  link: string
+  copy: JSX.Element
+  disclaimer?: string | JSX.Element
+  onDismissPromo: (event: React.MouseEvent<HTMLDivElement>) => void
```

Comment:
The user of this component ideally shouldn't have to deal with the event cancelling part, and so this callback should probably be `onDismissPromo: () => void`.

Can this component do the `preventDefault` thing itself?

Reply:
Yes, fixed.